### ISSUE TITLE
Vimeo Iframe getting block by RESPONSE-980-CORRELATION.conf rules

### ISSUE THREAD

**Issue by surajy93** [SpiderLabs/owasp-modsecurity-crs#1323]
*CRS version (e.g. v3.0.2):
*ModSecurity version (e.g. 2.9.2):
* Web Server and version (e.g. apache 2.4.27):
* Operating System and version:
 Access denied with code 403 (phase 2). Operator GE matched 5 at TX:anomaly_score. [file "/etc/modsecurity/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "93"] [id "949110"] [msg "Inbound Anomaly Score Exceeded (Total Score: 10)"] [severity "CRITICAL"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [ unique_id "XIeltwXBDmZrREXhmsoNiQAAAAM"], 

 [:error] [pid 16631] [client 103.58.8.207:27421] [client 103.58.8.207] ModSecurity: Warning. Operator GE matched 5 at TX:inbound_anomaly_score. [file "/etc/modsecurity/rules/RESPONSE-980-CORRELATION.conf"] [line "86"] [id "980130"] [msg "Inbound Anomaly Score Exceeded (Total Inbound Score: 10 - SQLI=0,XSS=10,RFI=0,LFI=0,RCE=0,PHPI=0,HTTP=0,SESS=0): NoScript XSS InjectionChecker: HTML Injection; individual paranoia level scores: 10, 0, 0, 0"] [tag "event-correlation"]  [unique_id "XIeltwXBDmZrREXhmsoNiQAAAAM"], 

[Tue Mar 12 12:35:48.847441 2019] [:error] [pid 16817] [client 206.189.133.105:60950] [client 206.189.133.105] ModSecurity: Request body no files data length is larger than the configured limit (131072).. Deny with code (413)   [unique_id "XIenpAcTE8Cv-I@z3cLpbwAAAAc"],

[Tue Mar 12 12:41:46.627632 2019] [:error] [pid 17229] [client 103.58.8.207:27367] [client 103.58.8.207] ModSecurity: Warning. detected XSS using libinjection. [file "/etc/modsecurity/rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf"] [line "60"] [id "941100"] [msg "XSS Attack Detected via libinjection"] [data "Matched Data: XSS data found within ARGS:_lp_lesson_video_intro: <iframe src=\\x22https://player.vimeo.com/video/322230047\\x22 width=\\x22640\\x22 height=\\x22360\\x22 frameborder=\\x220\\x22 webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>"] [severity "CRITICAL"] [ver "OWASP_CRS/3.1.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-xss"] [tag "OWASP_CRS/WEB_ATTACK/XSS"] [tag "WASCTC/WASC-8"] [tag "WASCTC/WASC-22"] [tag "OWASP_TOP_10/A3"] [tag "OWASP_AppSensor/IE1"] [tag "CAPEC-242"] [hostname "] [uri " [unique_id "XIepCme3dSYCFWQA7DRk8AAAABE"], 

[Tue Mar 12 12:41:46.631064 2019] [:error] [pid 17229] [client 103.58.8.207:27367] [client 103.58.8.207] ModSecurity: Warning. Pattern match "(?i)<[^\\\\w<>]*(?:[^<>\\"'\\\\s]*:)?[^\\\\w<>]*(?:\\\\W*?s\\\\W*?c\\\\W*?r\\\\W*?i\\\\W*?p\\\\W*?t|\\\\W*?f\\\\W*?o\\\\W*?r\\\\W*?m|\\\\W*?s\\\\W*?t\\\\W*?y\\\\W*?l\\\\W*?e|\\\\W*?s\\\\W*?v\\\\W*?g|\\\\W*?m\\\\W*?a\\\\W*?r\\\\W*?q\\\\W*?u\\\\W*?e\\\\W*?e|(?:\\\\W*?l\\\\W*?i\\\\W*?n\\\\W*?k|\\\\W*?o\\\\W*?b\\\\W*?j\\\\W*?e\\ ..." at ARGS:_lp_lesson_video_intro. [file "/etc/modsecurity/rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf"] [line "219"] [id "941160"] [msg "NoScript XSS InjectionChecker: HTML Injection"] [data "Matched Data: <iframe  found within ARGS:_lp_lesson_video_intro: <iframe src=\\x22https://player.vimeo.com/video/322230047\\x22 width=\\x22640\\x22 height=\\x22360\\x22 frameborder=\\x220\\x22 webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>"] [severity "CRITICAL"] [ver "OWASP_CRS/3.1.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-xss"] [tag "OWASP_CRS/WEB_ATTACK/XSS"] [tag "WASCTC/WASC-8"] [tag "WASCTC/WASC-22"] [tag "OWASP_TOP_10/A3"] [tag "OWASP_AppSensor/IE1"] [tag "CAPEC-242"] 

[Tue Mar 12 12:41:46.652400 2019] [:error] [pid 17229] [client 103.58.8.207:27367] [client 103.58.8.207] ModSecurity: Access denied with code 403 (phase 2). Operator GE matched 5 at TX:anomaly_score. [file "/etc/modsecurity/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "93"] [id "949110"] [msg "Inbound Anomaly Score Exceeded (Total Score: 10)"] [severity "CRITICAL"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"]  [unique_id "XIepCme3dSYCFWQA7DRk8AAAABE"], 

[Tue Mar 12 12:41:46.652692 2019] [:error] [pid 17229] [client 103.58.8.207:27367] [client 103.58.8.207] ModSecurity: Warning. Operator GE matched 5 at TX:inbound_anomaly_score. [file "/etc/modsecurity/rules/RESPONSE-980-CORRELATION.conf"] [line "86"] [id "980130"] [msg "Inbound Anomaly Score Exceeded (Total Inbound Score: 10 - SQLI=0,XSS=10,RFI=0,LFI=0,RCE=0,PHPI=0,HTTP=0,SESS=0): NoScript XSS InjectionChecker: HTML Injection; individual paranoia level scores: 10, 0, 0, 0"] [tag "event-correlation"] [hostname  [unique_id "XIepCme3dSYCFWQA7DRk8AAAABE"], 

 ModSecurity: Warning. Pattern match "^[\\\\d.:]+$" at REQUEST_HEADERS:Host. [file "/etc/modsecurity/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "798"] [id "920350"] [msg "Host header is a numeric IP address"] [data "206.189.133.105"] [severity "WARNING"] [ver "OWASP_CRS/3.1.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "OWASP_CRS/PROTOCOL_VIOLATION/IP_HOST"] [tag "WASCTC/WASC-21"] [tag "OWASP_TOP_10/A7"] [tag "PCI/6.5.10"] [hostname "206.189.133.105"] [uri "/qa.php"] [unique_id "XIYoUES9Fr3HBdQQpLIsWQAAAAs"]

Second issue : 

 ModSecurity: APR compiled version="1.6.2"; loaded version="1.6.3"
how to solve this.

@dune73 let me know anything else if you need

**Comment by dune73** [comment]
The request is being blocked by rule 949110, the standard rule blocking incoming requests after several rules triggered an alert.

Unfortunately, you did not provide those rules. No need to do a screen shot, copy pasting the alert message is good enough. And please be careful: Your screenshot includes your domain name.

**Comment by surajy93** [comment]
@dune73 I am a newbie for ModSecurity, using Wordpress how can I set rules for this. thanks.

**Comment by dune73** [comment]
You just need to send us the complete log. You can grep for the unique_id in the screenshot above.

**Comment by surajy93** [comment]
@dune73 okay.
![log](https://user-images.githubusercontent.com/42316893/54071955-12d73000-429a-11e9-913b-457fa6ba143e.PNG)
Log file

**Comment by surajy93** [comment]
@dune73 
the post blog is working fine but when posting iframe embed link from Vimeo getting blocked

**Comment by surajy93** [comment]
![block](https://user-images.githubusercontent.com/42316893/54072984-b4b14980-42a7-11e9-9893-aa10c43587b0.PNG)
need help tired of these error and guide to remove rules

**Comment by theMiddleBlue** [comment]
confirmed, PL1:

```
[941100] XSS Attack Detected via libinjection
[941160] NoScript XSS InjectionChecker: HTML Injection

Matched Data: XSS data found within ARGS:content: <iframe src="https://player.vimeo.com/video/167411059" width="640" height="480" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>
<p><a href="https://vimeo.com/167411059">[mirrors] - Hacking Sindicat de Mossos d'Esquadra (Catalan Police Union)</a> from <a href="https://vimeo.com/tukuls">TuKuLs</a> on <a href="https://vimeo.com">Vimeo</a>.</p>
```

It should be excluded by `9002700` but it doesn't fires any of the following tags:

```
[tag "application-multi"]
[tag "language-multi"]
[tag "platform-multi"]
[tag "attack-xss"]
[tag "OWASP_CRS/WEB_ATTACK/XSS"]
[tag "WASCTC/WASC-8"]
[tag "WASCTC/WASC-22"]
[tag "OWASP_TOP_10/A3"] 
[tag "OWASP_AppSensor/IE1"] 
[tag "CAPEC-242"]
```

What about adding `ctl:ruleRemoveTargetByTag=attack-xss;ARGS:content` to `9002700`?

**Comment by theMiddleBlue** [comment]
uhm... to be honest, I don't know if we should allow something like this...

**Comment by surajy93** [comment]
> uhm... to be honest, I don't know if we should allow something like this...

Thanks for the reply. 
 I will try maybe it works.

Refer : https://www.modsecurity.org/CRS/Documentation/anomaly.html.

Hope team will these solve soon.

**Comment by dune73** [comment]
A tail won't do. We need the complete requst in the log.

If you really do not know how this works, why don't you take a look at the tutorials at http://netnea,com.

**Comment by spartantri** [comment]
Allowing iframe injections is generally a very bad idea and is often abused to perform XSS attacks, if those are required it will be better to white list the allowed targets but be aware that this could open a big hole on the anti-XSS protection.

**Comment by surajy93** [comment]
@spartantri agreed with you. my website is a learning management system and it won't allow me to add a new course that is a big problem for me.
Don't want to white list looking for alternative solution.

**Comment by surajy93** [comment]
ModSecurity: Warning. detected XSS using libinjection. [file "/etc/modsecurity/rules/REQUEST-941-APPLICATION-ATTACK-XSS.conf"] [line "60"] [id "941100"] [msg "XSS Attack Detected via libinjection"] [data "Matched Data: XSS data found within ARGS:_lp_lesson_video_intro: <iframe src=\\x22https://player.vimeo.com/video/322230047\\x22 width=\\x22640\\x22 height=\\x22360\\x22 frameborder=\\x220\\x22 webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe>"] [severity "CRITICAL"] [ver "OWASP_CRS/3.1.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-xss"] [tag "OWASP_CRS/WEB_ATTACK/XSS"] [tag "WASCTC/WASC-8"] [tag "WASCTC/WASC-22"] [tag "OWASP_TOP_10/A3"] [tag "OWASP_AppSensor/IE1"] [tag "CAPEC-242"]  [unique_id "XIeltwXBDmZrREXhmsoNiQAAAAM"],

**Comment by spartantri** [comment]
In that case why don't you add a second vhost for accessing the site that use stronger authentication (2FA) to make sure only really trusted users can access it and set the xss inspections on that field

`SecRuleUpdateTargetByTag "attack-xss" !ARGS:contentARGS:content`

This will allow you to insert XSS content, so consider this risky and make it available to a controlled set of users and sources.

I'm closing this but feel free to post any other concern or contact us via slack https://owasp.slack.com in the channel #coreruleset.

If you are not yet on the OWASP Slack, here is your invite: https://join.slack.com/t/owasp/shared_invite/enQtNDI5MzgxMDQ2MTAwLTEyNzIzYWQ2NDZiMGIwNmJhYzYxZDJiNTM0ZmZiZmJlY2EwZmMwYjAyNmJjNzQxNzMyMWY4OTk3ZTQ0MzFhMDY
